### PR TITLE
feat: (WIP/Sketch) add TVL to ProtocolSim

### PR DESCRIPTION
**Pull request by tamaralipows**
Questions:
- We actually have the TVL inside Tycho - I'm assuming - should we maybe expose that inside of `ComponentWithState` instead of re-deriving it from the balances?
- Should we give TVL in ETH? Or the quote token? I would assume in ETH. Where then, do we get the ETH price? For now, I've started designing this using the quote token...

### CODE REVIEW

File: examples/explorer/main.rs [hunk 1]
```diff
@@ -36,32 +36,21 @@ async fn main() {
...
+    let consumer_handle = task::spawn(async move {
+        while let Some(message) = tick_rx.recv().await {
+            // println!("got message: {:?}", message);
+        }
+        Ok(())
     });
-    let tasks = [tycho_message_processor, terminal_app];
+    let tasks = [tycho_message_processor, consumer_handle];
+
     let _ = select_all(tasks).await;
```
Comment (tamaralipows) [inline examples/explorer/main.rs:55]: Accidental commit. Revert later.

**Comment by kayibal** [comment]
Do we really need this interface change? What is the use case of having tvl exposed on the main interface?

**Comment by dianacarvalho1** [comment]
> Do we really need this interface change? What is the use case of having tvl exposed on the main interface?

This was brought up by Markus during Edge City. That his first application would be to filter/order the pools by TVL..

**Comment by tamaralipows** [comment]
Closing this for two reasons:
1.  It's not in the docs so it seems very non-essential.
2.  It does not seem like an easy thing to do. It also requires an interface change - so it's not a small issue.

**Pull request closed without merging**